controllers/util/solr_util.go

```diff
@@ -971,10 +991,26 @@ func generateZKInteractionInitContainer(solrCloud *solr.SolrCloud, solrCloudStat
 		})
 	}
 
+	cmd := ""
+
 	if solrCloud.Spec.SolrTLS != nil {
-		cmd := "solr zk ls ${ZK_CHROOT} -z ${ZK_SERVER} || solr zk mkroot ${ZK_CHROOT} -z ${ZK_SERVER}" +
+		cmd = "solr zk ls ${ZK_CHROOT} -z ${ZK_SERVER} || solr zk mkroot ${ZK_CHROOT} -z ${ZK_SERVER}" +
 			"; /opt/solr/server/scripts/cloud-scripts/zkcli.sh -zkhost ${ZK_HOST} -cmd clusterprop -name urlScheme -val https" +
 			"; /opt/solr/server/scripts/cloud-scripts/zkcli.sh -zkhost ${ZK_HOST} -cmd get /clusterprops.json"
+	}
+
+	volumeMounts := []corev1.VolumeMount{}
+	if configMapInfo[SecurityJsonFile] != "" {
+		// mount security.json from the ConfigMap at /tmp/security.json
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{Name: "security-json", MountPath: "/tmp"})
+		if cmd == "" {
+			cmd += "solr zk ls ${ZK_CHROOT} -z ${ZK_SERVER} || solr zk mkroot ${ZK_CHROOT} -z ${ZK_SERVER}; "
+		}
+		cmd += "ZK_SECURITY_JSON=$(/opt/solr/server/scripts/cloud-scripts/zkcli.sh -zkhost ${ZK_HOST} -cmd get /security.json); "
+		cmd += "if [ ${#ZK_SECURITY_JSON} -lt 3 ]; then /opt/solr/server/scripts/cloud-scripts/zkcli.sh -zkhost ${ZK_HOST} -cmd putfile /security.json /tmp/security.json; echo \"put security.json in ZK\"; fi"
```

Comment:
haha ... don't laugh at this ;-) it was the easiest way to check if the znode data is empty "" or {} using /bin/sh